File: 2018/1/luke1.js
const fs = require("fs");

const numbers = fs
    .readFileSync("input-vekksort.txt", "UTF-8")
    .split("\n")
    .map(Number);

function vekkSort(arr) {
    const res = [];
    let prev = -1;
    for (elem of arr) {
        if (elem >= prev) {
            res.push(elem);
            prev = elem;
        }
    }

    return res;
}
const sorted = vekkSort(numbers);
const res = sorted.reduce((acc, curr) => acc + curr, 0);

console.log(res);

File: 2018/1/luke1.py

with open("input-vekksort.txt") as f:
    lines = f.readlines()
    
    sum = 0
    prev = 0

    for elem in lines:
        elemParsed = int(elem)
        if(elemParsed < prev):
            continue
        sum += elemParsed
        prev = elemParsed

print(sum)